src/targets/gpu/include/migraphx/gpu/convolution.hpp

```diff
@@ -24,20 +24,60 @@
 #ifndef MIGRAPHX_GUARD_RTGLIB_CONVOLUTION_HPP
 #define MIGRAPHX_GUARD_RTGLIB_CONVOLUTION_HPP
 
+#include "migraphx/op/identity.hpp"
 #include <migraphx/shape.hpp>
-#include <migraphx/op/convolution.hpp>
+#include <migraphx/generate.hpp>
+#include <migraphx/register_op.hpp>
 #include <migraphx/gpu/miopen.hpp>
-
+#include <migraphx/op/convolution.hpp>
+#include <migraphx/op/quant_convolution.hpp>
+#include <migraphx/op/deconvolution.hpp>
+#include <unordered_map>
+#include <migraphx/reflect.hpp>
+#include <migraphx/gpu/context.hpp>
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace gpu {
 
-struct context;
+inline shape reshape_if_1d(const shape& input)
+{
+    shape new_shape{input};
+    auto dims = new_shape.lens();
+
+    if(dims.size() == 3)
+    {
+        std::vector<size_t> new_dims = dims;
+        new_dims.insert(new_dims.begin() + 2, 1);
+        new_shape = shape{input.type(), new_dims};
+    }
+    return new_shape;
+}
+
+template <typename T>
+auto conv_compute_shape(rank<2>, const T& op, const std::vector<shape>& inputs)
+    -> decltype(op.normalize_compute_shape(inputs))
+{
+    check_shapes{inputs, op}.has(4).standard();
+    std::vector<shape> conv_inputs(inputs.begin(), inputs.begin() + 2);
+    check_shapes{conv_inputs, op}.max_ndims(5);
+    return op.normalize_compute_shape(conv_inputs);
+}
+
+template <typename T>
+auto conv_compute_shape(rank<1>, const T& op, const std::vector<shape>& inputs)
+    -> decltype(op.compute_shape(inputs))
+{
+    check_shapes{inputs, op}.has(4).standard();
+    std::vector<shape> conv_inputs(inputs.begin(), inputs.begin() + 2);
+    check_shapes{conv_inputs, op}.max_ndims(5);
+    return op.compute_shape(conv_inputs);
+}
 
 struct miopen_convolution
 {
-    op::convolution op;
+    operation op                      = op::identity{};
```

Comment:
Any reason to default this to identity instead of convolution?

Reply:
Removed.

Reply:
Identity is needed. It requires some reference to operation object without that it  fails on assertion in `operation.hpp`.

Reply:
This is similar to how it is done for gemms and rocblas.

Reply:
Made changes to make it templated.